clients/src/main/java/org/apache/kafka/common/config/ConfigDef.java

```diff
@@ -1006,6 +1036,9 @@ public String toEnrichedRst() {
 
         String lastKeyGroupName = "";
         for (ConfigKey key : sortedConfigs()) {
+            if (key.deprecated) {
+                continue;
```

Comment:
I think marking it as deprecated in html would be better than skipping it. So if #2456 is definitely going in we can close this PR then.